src/main/java/org/zalando/putittorest/HttpAsyncClientFactoryBean.java

```diff
@@ -24,10 +29,19 @@ public void setLastResponseInterceptors(final List<HttpResponseInterceptor> inte
         interceptors.forEach(builder::addInterceptorLast);
     }
 
+    @Hack("Hide type to avoid runtime dependency")
+    public void setMetricsWrapper(final Object metricsWrapper) {
+        this.metricsWrapper = metricsWrapper;
+    }
+
     @Override
     public HttpAsyncClient getObject() {
         // TODO: builder.setConnectionTimeToLive(30, TimeUnit.SECONDS);
-        return builder.build();
+        final CloseableHttpAsyncClient client = builder.build();
+        if (metricsWrapper != null) {
+            return new InstrumentedHttpAsyncClient((MetricsWrapper) metricsWrapper, client);
```

Comment:
Does this typecast work when the dependency is not available, even if it is inside an `if`? If the type would only be used inside `InstrumentedHttpAsyncClient` it would definitely be safe.